File: 2024/April/4-4 Maximum Nesting Depth of the Parentheses/MaxNestingDepth.py
class Solution:
    def maxDepth(self, s: str) -> int:
        depth = 0
        maxDepth = 0

        for c in s:
            if c == "(":
                depth += 1
                if depth > maxDepth:
                    maxDepth = depth
            elif c == ")":
                depth -= 1
        
        return maxDepth

File: 2024/April/4-4 Maximum Nesting Depth of the Parentheses/MaxNestingDepth.java
class Solution {
    public int maxDepth(String s) {
        int depth = 0;
        int max = 0;
        for(char c : s.toCharArray()) {
            if(c == '(') {
                depth++;
                if(depth > max) {
                    max = depth;
                }
            }
            else if(c == ')') {
                depth--;
            }
        }

        return max;
    }
}